test/tftp/tftp_server_test.rb

```diff
@@ -0,0 +1,146 @@
+require 'test_helper'
+require 'tftp/server'
+require 'tftp/tftp_plugin'
+require 'tempfile'
+
+module TftpGenericServerSuite
+  def setup
+    @rootdir = "/some/root"
+    @mac = "aa:bb:cc:dd:ee:ff"
+    @content = "file content"
+    Proxy::TFTP::Plugin.settings.stubs(:tftproot).returns(@rootdir)
+    setup_paths
+  end
+
+  def pxe_config_files
+    @pxe_config_files.collect{|f| File.join(@rootdir, f)}
+  end
+
+  def pxe_default_files
+    @pxe_default_files.collect{|f| File.join(@rootdir, f)}
+  end
+
+  def test_set
+    pxe_config_files.each do |file|
+      @subject.expects(:write_file).with(file, @content).once
+    end
+    @subject.set @mac, @content
+  end
+
+  def test_del
+    pxe_config_files.each do |file|
+      @subject.expects(:delete_file).with(file).once
+    end
+    @subject.del @mac
+  end
+
+  def test_get
+    file = pxe_config_files.first
+    @subject.expects(:read_file).with(file).returns(@content)
+    assert_equal @content, @subject.get(@mac)
+  end
+
+  def test_create_default
+    pxe_default_files.each do |file|
+      @subject.expects(:write_file).with(file, @content).once
+    end
+    @subject.create_default @content
+  end
+end
+
+class HelersServerTest < Test::Unit::TestCase
+  def setup
+    @subject = Proxy::TFTP::Server.new
+  end
+
+  def test_path_with_settings
+    Proxy::TFTP::Plugin.settings.expects(:tftproot).returns("/some/root")
+    assert_equal "/some/root", @subject.send(:path)
+  end
+
+  def test_path
+    assert_match /file.txt/, @subject.send(:path, "file.txt")
```

Comment:
Here, and other protected methods -- dropping 'protected' would allow you to call methods directly instead of relying on 'send'.